src/api/fetchAPI_rest.ts

```diff
@@ -86,6 +86,87 @@ export class fetchAPI_rest implements fetchAPI {
         .then((res) => res.json());
     }
 
+    getDescription = async (
+        elementType: string | undefined,
+        elementName: string | undefined,
+        tenant: string,
+        repo: string,
+        env: string,
+        version: string | undefined,
+    ) => {
+        const filename = `${elementType}/${elementName}.md`;
+        const response = await this.getDescriptionFile(filename, tenant, repo, env, version);
+        const responseBody = await response.json();
+        if (!response.ok) {
+            throw new Error(responseBody["detail"]);
+        }
+        return await this.resolveDescriptionImageUrls(responseBody.content, tenant, repo, env, version);
+    };
+
+    /**
+     * For description .md files fetched from REST endpoint, images with relative URLs 
+     * will be fetched from the same endpoint and their URLs will be resolved to blob URLs
+     * 
+     * E.g. On local, `![less flights more ...](images/train.png)`
+     * will be fetched from `/api/v1/descriptions/images/train.png?tenant={tenant}&repo={repo}&env={env}&version={version}`
+     * and will be resolved to `![less flights more ...](blob:http://localhost:5173/fe475b95-c790-4739-8169-ff09e3696462)`
+     */
+    private resolveDescriptionImageUrls = async (
+        markdown: string,
+        tenant: string,
+        repo: string,
+        env: string,
+        version: string | undefined
+    ) => {
+        // Matches "![less flights more ...](images/train.png)"
+        // Does not match
+        // - "![Lorem Picsum](https://picsum.photos/200/300)"
+        // - "![Lorem Picsum](http://picsum.photos/200/300)"
+        // - "![Lorem Picsum](ftp://picsum.photos/200/300)"
+        const regex = /(!\[.*?\]\()((?!https?|ftp).*)(\))/g;
+        const promises: Array<Promise<any>> = [];
+        const fileUrlMap: Map<string, string> = new Map();
+
+        // Fetch and resolve all URLs
+        const matches = markdown?.matchAll(regex);
+        for (const match of matches) {
+             // First capture group: "![less flights more ...]("
+             // Second capture group: "images/train.png"
+             // Last capture group: ")"
+            const filename = match[2];
+            const promise = this.getDescriptionFile(filename, tenant, repo, env, version).then((response) => {
```

Comment:
> Interesting approach to replace relative Urls. I did not yet think about that...
> 
> Maybe we could also use the following feature of react-markup instead - it has a transformLinkUri and transformImageUri option, see https://github.com/remarkjs/react-markdown/tree/8.0.3.
> 
> So the fetchAPI would need another interface method getDescriptionUriTransform, which would be given as option to Markdown component in src/components/ConfigExplorer/DescriptionTab.tsx.
> 
> WDYT?

This question is a bit tricky to answer after some researching

First, for context:
- With our endpoint `/descriptions/{filename}`, since the resources are not publicly available, we cannot have simple URLs directly linked to the images in our markdown files
- We have to make async request calls and have to have authentication with Bearer Token to the endpoint, which means dynamically fetching the images and resolving the URLs before rendering

What I have done:
- I have tried out `transformLinkUri` and `transformImageUri` in `react-markdown`, but they are synchronous! I have tried some workarounds but none worked :( I could not find anywhere that they support asynchronous functions. This is their default [uriTransformer](https://github.com/remarkjs/react-markdown/blob/8.0.3/lib/uri-transformer.js), which is also synchronous
- I had a quick look through the list of plugins ([`remark`](https://github.com/remarkjs/remark/blob/main/doc/plugins.md#list-of-plugins) & [`rehype`](https://github.com/rehypejs/rehype/blob/HEAD/doc/plugins.md#list-of-plugins)) we could use, but couldn't find anything

So in my opinion, our options right now are:
1. Write our own `remark` plugin to make async calls. Not sure if [this will work](https://github.com/remarkjs/react-markdown/issues/680) or even worth it. [This could help with inspiration if we decide to](https://github.com/treeverse/lakeFS/pull/7745)
2. We can keep the current implementation. The logic for fetching and resolving images URLs is abstracted away in `fetchAPI` and used specifically for description files anyway. We just need to document the convention for using images URLs in our description markdown files.

WDYT? Please let me know if you have any suggestion about what direction we should take

Reply:
> Why is it needed to prefetch all links here?

For the images to be rendered, they all need to be prefetched and converted to blob URLs in the browser first. Or else `react-markdown` cannot understand the absolute links in our markdowns (e.g. `images/train.png`)

And since I also cannot use `transformLinkUri` and `transformImageUri`, there is no way to intercept and dynamically fetch images when the markdown is being rendered. 

Please let me know if you have other ideas

Reply:
Ok, i understand the problem now.
You found a good solution then.
Let's merge.